arkouda/pdarraycreation.py

```diff
@@ -512,40 +525,15 @@ def linspace(start : int, stop : int, length : int) -> pdarray:
     >>> ak.linspace(start=-5, stop=0, length=5)
     array([-5, -3.75, -2.5, -1.25, 0])
     """
-    if not all((np.isscalar(start), np.isscalar(stop), np.isscalar(length))):
-        raise TypeError("all arguments must be scalars")
-
-    starttype = resolve_scalar_dtype(start)
-
-    try: 
-        startstr = NUMBER_FORMAT_STRINGS[starttype].format(start)
-    except KeyError as ke:
-        raise TypeError(('The start parameter must be an int or a scalar that'  +
-                        ' can be parsed to an int, but is a {}'.format(ke)))
-    stoptype = resolve_scalar_dtype(stop)
-
-    try: 
-        stopstr = NUMBER_FORMAT_STRINGS[stoptype].format(stop)
-    except KeyError as ke:
-        raise TypeError(('The stop parameter must be an int or a scalar that'  +
-                        ' can be parsed to an int, but is a {}'.format(ke)))
-
-    lentype = resolve_scalar_dtype(length)
-    if lentype != 'int64':
-        raise TypeError("The length parameter must be an int64")
-
-    try: 
-        lenstr = NUMBER_FORMAT_STRINGS[lentype].format(length)
-    except KeyError as ke:
-        raise TypeError(('The length parameter must be an int or a scalar that'  +
-                        ' can be parsed to an int, but is a {}'.format(ke)))
-
-    repMsg = generic_msg("linspace {} {} {}".format(startstr, stopstr, lenstr))
+    repMsg = generic_msg("linspace {} {} {}".format(start, stop, length))
     return create_pdarray(cast(str,repMsg))
```

Comment:
why is there a `cast` here? you should get a `str` response.

Reply:
@mhmerrill generic_msg returns str or bytes:

![image](https://user-images.githubusercontent.com/10785153/103571155-6582d280-4e98-11eb-9b62-ca544da0da0d.png)

Reply:
in this case it should only return a `str`. does it trip up the type checking? I ask because other places in the code don't have a `cast`

Reply:
@mhmerrill Looks like this cast is no longer needed. At one point it was as the mypy type check was failing, but not now. Good catch! I've removed the cast from the codebase.